baybe/surrogates/gaussian_process/presets/default.py

```diff
@@ -17,54 +19,39 @@
     from baybe.kernels.base import Kernel
     from baybe.searchspace.core import SearchSpace
 
+# Boundaries for low and high dimension limits
+_DIM_LIMITS = (8, 75)
+
 
 @define
 class DefaultKernelFactory(KernelFactory):
     """A factory providing the default kernel for Gaussian process surrogates.
 
-    The logic is adapted from EDBO (Experimental Design via Bayesian Optimization).
-
-    References:
-    *   https://github.com/b-shields/edbo
-    *   https://doi.org/10.1038/s41586-021-03213-y
+    This is taking the low and high dimensional limits of
+    :class:`baybe.surrogates.gaussian_process.presets.default.EDBOKernelFactory`
+    and interpolates the prior moments linearly between them.
     """
 
     def __call__(  # noqa: D102
         self, searchspace: SearchSpace, train_x: Tensor, train_y: Tensor
     ) -> Kernel:
         # See base class.
-
-        mordred = (searchspace.contains_mordred or searchspace.contains_rdkit) and (
-            train_x.shape[-1] >= 50
+        effective_dims = train_x.shape[-1] - len(
+            [p for p in searchspace.parameters if isinstance(p, TaskParameter)]
         )
 
-        # low D priors
-        if train_x.shape[-1] < 10:  # <-- different condition compared to EDBO
-            lengthscale_prior = GammaPrior(1.2, 1.1)
-            lengthscale_initial_value = 0.2
-            outputscale_prior = GammaPrior(5.0, 0.5)
-            outputscale_initial_value = 8.0
-
-        # DFT optimized priors
-        elif mordred and train_x.shape[-1] < 100:
-            lengthscale_prior = GammaPrior(2.0, 0.2)
-            lengthscale_initial_value = 5.0
-            outputscale_prior = GammaPrior(5.0, 0.5)
-            outputscale_initial_value = 8.0
-
-        # Mordred optimized priors
-        elif mordred:
-            lengthscale_prior = GammaPrior(2.0, 0.1)
-            lengthscale_initial_value = 10.0
-            outputscale_prior = GammaPrior(2.0, 0.1)
-            outputscale_initial_value = 10.0
-
-        # OHE optimized priors
-        else:
-            lengthscale_prior = GammaPrior(3.0, 1.0)
-            lengthscale_initial_value = 2.0
-            outputscale_prior = GammaPrior(5.0, 0.2)
-            outputscale_initial_value = 20.0
+        # Interpolate prior moments linearly between low D and high D regime
+        # Values outside the dimension limits will get the border value assigned
+        lengthscale_prior = GammaPrior(
```

Comment:
I'm not sure from where exactly you took the upper values. Seem like average of mordred and OHE, is that correct? If so, could you add corresponding comments here and to the noise factory?

Reply:
exactly right, added